### ISSUE TITLE
Multisite problems

### ISSUE THREAD

**Issue by dbeja** [awslabs/aws-refarch-wordpress#23]
I launched the stack and everything was working fine, but then I activated multisite option and follow the Wordpress network setup I can't open the site anymore.
When I try to login wp-login.php is fine and 200 but it also tries to load load-styles.php directly from one of the private ips that is not even up anymore.
It was the ip I used to ssh from bastion to change wp-config.
Any ideas what could be the problem?

I also checked issue #14 and added more ports to alb health check but didn't fix the problem.

Thanks

**Comment by blandman** [comment]
I know this is old, but I was able to get Multisite working by updating the _blogs and all _options mysql tables to reflect the DNS of the ELB. This is somewhat problematic when trying to reverse proxy, but I'll post the nginx configuration when I have it working. Should be possible with cloudfront as well, as long as w3-total-cache is configured to minify and serve all js/css files from s3/cloudfront.

**Comment by fakekamrankhan** [comment]
@blandman does your configuration work with top-level domains? I want to use this CloudFormation stack to host a multisite WP that supports multiple domains and the documentation isn't very clear on whether this will work with the existing set of templates.

**Comment by blandman** [comment]
It should work on top level domains, did not configure it for subdomains but should be manually configurable.

**Comment by fakekamrankhan** [comment]
Are you going to commit those changes to the repository or are they already in there?

I was consulting [this guide in the WP codex](https://codex.wordpress.org/WordPress_Multisite_Domain_Mapping) that indicates top-level domains are now inherently a part of WP since 4.9 and shows some basic setup for getting SSLs configured with SNI but I have yet to test. Would love to know if you have done this already.

**Comment by fakekamrankhan** [comment]
That sounds great! Look forward to your changes.

**Comment by blandman** [comment]
I've made the push here: https://github.com/psd401/aws-refarch-wordpress
Pull that repo and upload to S3 for your template, works with restoring from previous EFS and Database, though with Multisite still manual addition of Multisite wp-config options is still required.

**Comment by ecout** [comment]
You probably have to modify the template and redeploy the stack from scratch as indeed once you get into multisite the domain becomes a blocker in the database and the site will redirect you, even if you get it working. 
You want to CREATE the database with multisite from scratch with the respective wp-config settings, otherwise you have to do what @blandman said here:
I know this is old, but I was able to get Multisite working by updating the _blogs and all _options mysql tables to reflect the DNS of the ELB.
Once you do that if you do a gray box test:
Isolate a single instance and hit its ip on multisite mode WITHOUT specifying the HTTP header that tells wordpress where you're coming from(your site FQDN). The wp instance will redirect you to the ELB/ALB thru the Route53 record that is hardcoded on the Multisite database and this makes things tightly coupled.
In plain English, wordpress is not designed for this kind of stuff so you have to work around it. 

Old CMS environment like wordpress are not designed to be throw away and replace like we do in the Cloud era and thus are tightly coupled so you're bound to run into this kind of trouble.

That's why you use blue/green deployments.
